### PR TITLE
Check controller_interface::init return value when loading

### PR DESCRIPTION
**Pull request by AugusteBourgois**
solves #383

### CODE REVIEW

File: controller_manager/test/test_controller_failed_init/test_controller_failed_init.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
```
Comment (destogl) [inline controller_manager/test/test_controller_failed_init/test_controller_failed_init.cpp:1]: Please update the copyright to your company name or "ros2_control development team" if you like. And also the year.

**Review by destogl — COMMENTED** [review]
Does this look great thanks? What do you think about integrating this into [`test_load_controller`](https://github.com/ros-controls/ros2_control/blob/master/controller_manager/test/test_load_controller.cpp) test file? The test on its own is relatively short and putting it there could save us some additional files/code.

**Comment by AugusteBourgois** [comment]
If I understood correctly, you would like me to merge the extra cpp files into a single short class at the beginning of the test_load_controller file, right ?

**Comment by destogl** [comment]
> If I understood correctly, you would like me to merge the extra cpp files into a single short class at the beginning of the test_load_controller file, right ?

For example. Or just to merge the test there. As I understood you don't need an additional TestController implementation, but you can use the existing one. You are testing is the proper controller name is given, am I right? So, you need at least one test controller, but probably the existing one can help. 

Do you really need an additional test class? Or is it possible to use an existing one?

**Comment by AugusteBourgois** [comment]
The feature added is basically to return a `nullptr` instead of a Controller interface shared pointer when the `init` function returns an `return_type::ERROR` error code.
So I need a `TestControler` implementation with an `init` function returning that code.
I could indeed overload the `init` function of an existing controller for the sake of conciseness, if you like.

**Comment by AugusteBourgois** [comment]
> The feature added is basically to return a `nullptr` instead of a Controller interface shared pointer when the `init` function returns an `return_type::ERROR` error code.
> So I need a `TestControler` implementation with an `init` function returning that code.
> I could indeed overload the `init` function of an existing controller for the sake of conciseness, if you like.

I don't think I can actually do that. When calling `cm_->load_controller(...)`, the controller is loaded through pluginlib by class name, so I need a class with a specific name, that has been referenced in a specific xml file.

Do you have any idea on how I could tackle this ?

I thought about merging both `test_controller` and `test_controller_failed_init` folders and files (one cpp/hpp/xml file for both, since the `failed_init` version simply overrides the `init` function of `test_controller`).

File: controller_manager/test/test_controller_failed_init/test_controller_failed_init.cpp (hunk 1 above)
Comment (bmagyar) [reply controller_manager/test/test_controller_failed_init/test_controller_failed_init.cpp:1]:
```suggestion
// Copyright 2020 ros2_control development team
```

File: controller_manager/test/test_controller_failed_init/test_controller_failed_init.hpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
```
Comment (bmagyar) [inline controller_manager/test/test_controller_failed_init/test_controller_failed_init.hpp:1]:
```suggestion
// Copyright 2020 ros2_control development team
```

**Comment by destogl** [comment]
@AugusteBourgois thanks for trying this. You are right, I see now the issue. I am OK with your approach.

**Review by bmagyar — APPROVED** [review]

**Review by destogl — APPROVED** [review]

**Pull request merged**